### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by MuhamedEssam**
i hope this time is closer

### CODE REVIEW

File: tests/client/test_machine.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from testutils import ClientTest
+
+Dense = tc.tensor.Dense
+ENDPOINT = "/transact/hypothetical"
+LEARNING_RATE = tc.F32(0.01)
+MAX_ITERATIONS = 10
+NUM_EXAMPLES = 20
+
+
+class Logistic_Regression():
```
Comment (haydnv) [inline tests/client/test_machine.py:14]: I still don't see the advantage of making this a class. You can just delete this line and then it will be a function.

File: tests/client/test_machine.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+               [2,-10,0],
+               [10,-5,1],
+               [1,-3,0],
+               [6,-5,1],
+               [2,-7,0],
+               [6,-1,1]])
+        X=data[:,:-1]
+        y=data[:,-1]
+        
+        X_train,X_test,y_train,y_test=train_test_split(X,y,random_state=0,test_size=0.2)
```
Comment (haydnv) [inline tests/client/test_machine.py:45]: rather than add a new dependency on SciKit (even just in the tests) I think it would just be better to say
```python
pivot = len(data) // 2
X = data[:pivot]
y = data[pivot:]
# ...
```

File: tests/client/test_machine.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+               [6,-5,1],
+               [2,-7,0],
+               [6,-1,1]])
+        X=data[:,:-1]
+        y=data[:,-1]
+        
+        X_train,X_test,y_train,y_test=train_test_split(X,y,random_state=0,test_size=0.2)       
+        cxt.inputs=load(X_train)
+        cxt.labels=load(y_train)        
+        cxt.result = Logistic_Regression.fit(cxt.inputs,cxt.labels)
```
Comment (haydnv) [inline tests/client/test_machine.py:48]: you didn't instantiate the class here (but again, I don't think you need a class)

File: tests/client/test_machine.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            return (i <= MAX_ITERATIONS).logical_and(((output > 0.5) != labels).any())
+        
+        def cost(self,output, labels, dl=False):
+            if dl:
+                return output.sub(labels).mul(2).mean()
+            return output.sub(labels).pow(2).mean()        
+        
+        # for trial only , Adam must be replaced by Gradient Descent !
+        optimizer = tc.ml.optimizer.Adam.create(param_list=logistic.get_param_list(), lr=LEARNING_RATE)
+        result= tc.ml.optimizer.train(logistic, optimizer, inputs, labels, cost, train_while)
```
Comment (haydnv) [inline tests/client/test_machine.py:28]: This doesn't work because `train` contains a `closure` that tries to capture states from the outer context, but there is no outer context here. I think you can fix this problem by making `fit` a `tc.post_op`.

**Review by haydnv — CHANGES_REQUESTED** [review]

**Comment by haydnv** [comment]
Superceded by #116 -- again, please commit updates to existing PRs when possible, and close any old PRs if you intend to merge changes from a new one instead
